utils/oscap-xccdf.c

```diff
@@ -1041,6 +1046,9 @@ bool getopt_xccdf(int argc, char **argv, struct oscap_action *action)
                 } else {
                     action->f_ovals = NULL;
                 }
+
+		if (action->rule != NULL && action->profile == NULL)
+			return oscap_module_usage(action->module, stderr, "Profile needs to be specified when evaluating a single rule!");
```

Comment:
@matusmarhefka "Default" profile is a virtual profile that is used when no profile is specified. You can trigger it if you run `oscap xccdf eval` without `--profile`.

If no profile is specified, OpenSCAP evaluates all  rules (`<Rule>`) with attribute `selected="true"`, regardless of which profile they belong.

The reason why it does nothing in your case is that in SSG content there is no Rule with attribute `selected="true"`, in other words, the  virtual "Default profile" contains 0 rules.

But in other content, there can be some of there rules. You can have also content with no profiles. For example this file has no profile, but it's a valid XCCDF: https://github.com/OpenSCAP/openscap/blob/maint-1.2/tests/API/XCCDF/unittests/test_xccdf_oval_relative_path.xccdf.xml